File: ga/4omini_result_2024_7/result/1968C_cpp_mini/40_1968C.cpp
use std::io;

const N: usize = 500;
const A: i64 = 1_000_000_000;

fn proc() {
    let mut x = vec![0; N];
    let mut a = vec![0; N];
    
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Not a valid number");

    for i in 1..n {
        input.clear();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        x[i] = input.trim().parse::<i64>().expect("Not a valid number");
    }

    a[0] = x[1] + 1;

    for i in 1..n {
        let mut j = 0;
        loop {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i]) || (a[i] != a[i - 1] && i + 1 == n) {
                break;
            }
            j += 1;
            if j > A {
                break; // safety break similar to C++ code limit
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    
    println!();
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: usize = input.trim().parse().expect("Not a valid number");

    for _ in 0..t {
        proc();
    }
}

File: ga/gpt3_5_result_2024_6/result/1968C_cpp_gpt/40_1968C.rs
use std::io;

const N: usize = 500;
const X: usize = 500;
const A: usize = 1000000000;

fn proc() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let x: Vec<usize> = input.split_whitespace().map(|s| s.parse().unwrap()).collect();

    let mut a: Vec<usize> = vec![0; n];
    a[0] = x[0] + 1;
    for i in 1..n {
        let mut j = 0;
        while j <= A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i] || i + 1 >= n) && a[i] != a[i - 1] {
                break;
            }
            j += 1;
        }
    }

    for val in a {
        print!("{} ", val);
    }
    println!();
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}